service/controller/v1/resource_set.go

```diff
@@ -129,7 +131,24 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 	}
 
 	handlesFunc := func(obj interface{}) bool {
-		return true
+		// The controller's informer is configured to watch services in all
+		// namespaces because tenant cluster namespaces can be any random string. We
+		// do not want to monitor kube-system specific services here, as this had
+		// ugly side effects in the past.
+		//
+		//     https://github.com/giantswarm/giantswarm/issues/5168
+		//
+		{
+			m, err := meta.Accessor(obj)
+			if err != nil {
+				config.Logger.Log("level", "error", "message", "failed parsing object meta", "stack", fmt.Sprintf("%#v", err))
+			}
+			if m.GetNamespace() != "kube-system" {
+				return true
+			}
+		}
```

Comment:
I'm afraid at some point we may want to monitor something from the kube-system. Then someone will lose a lot of time figuring out why the service isn't observed. I think the finalizer should be removed, this looks like a bug. I left my comment here: https://github.com/giantswarm/giantswarm/issues/5168#issuecomment-459281865.

I'm not going to block that, but I'm not sure if I should approve either.

Reply:
This project here is on life support. Team Magic will work on Monitoring improvements after VOO. Joe's comment was literally: "when we do not get rid of it with the prometheus-operator we did something very wrong".

Reply:
When Joe said so :D

Reply:
IIRC, this is the watch for the services that the prometheus-config-controller uses to determine which tenant clusters to watch.

It doesn't affect which namespaces are actually scraped by Prometheus.